fw/http.c

```diff
@@ -4121,12 +4122,13 @@ __tfw_http_resp_fwd(TfwCliConn *cli_conn, struct list_head *ret_queue)
 			return;
 		}
 		list_del_init(&req->msg.seq_list);
-		tfw_http_resp_pair_free(req);
 		TFW_INC_STAT_BH(serv.msgs_forwarded);
+		tfw_inc_global_hm_stats(req->resp->status);
+		tfw_http_resp_pair_free(req);
```

Comment:
Personally, I dislike that `tfw_http_resp_pair_free()` and `list_del_init()` are separated now by an aside logic. The are logically coupled and should stay together. That's why stats accounting is usually done after they are called in other places of the code. I can see that you need `req` alive in this particular place. Perhaps, stats accounting can be done before the call to `list_del_init()` in this specific case? I believe that was the code would be a little cleaner and easier to understand. It's not like it matters much from the functional point of view. Just an opinion. Thanks.

Reply:
I don't know, there are no much difference from my side. Anyway, fixed.